### PR TITLE
Refactor systemd unit installation

### PR DESCRIPTION
**Pull request by jtagcat**

### CODE REVIEW

**Comment by Foxboron** [comment]
While I don't object to the changes it would be nice to get a better description why these changes are needed.

They also seem to change some inconsequential function names for no particular reason?

**Comment by jtagcat** [comment]
[fix slog formatting](https://github.com/Foxboron/ssh-tpm-agent/pull/23/commits/f1338eb0ac8b617ba2214f5f53a0d5c18dec9833) is because slog doesn't use fmt (see https://pkg.go.dev/log/slog)

Well, the context is that I'm trying to get it working. Some (mainly spacing changes) are by auto-run gofmt.

For the other commit, I went looking why my user unit exec pointed to `$HOME/go/bin`, not `$HOME/go/bin/ssh-tpm-agent`. In the end, I have no idea why, it seems fine, when trying again.

The systemd check was wrong: you can run systemd without the `.config/systemd/user` directory existing.

I had to read the code. The function name changes are for readability, in most cases `Get*` is not needed, as it is assumed. https://www.youtube.com/watch?v=-J3wNP6u5YU might give a fair-enough longer explanation.

**Comment by Foxboron** [comment]
>For the other commit, I went looking why my user unit exec pointed to $HOME/go/bin, not $HOME/go/bin/ssh-tpm-agent. In the end, I have no idea why, it seems fine, when trying again.

This change probably; https://github.com/Foxboron/ssh-tpm-agent/commit/e1ecab37dacce5631b73ab2309e9e13e65acd964

**Comment by jtagcat** [comment]
> > For the other commit, I went looking why my user unit exec pointed to $HOME/go/bin, not $HOME/go/bin/ssh-tpm-agent. In the end, I have no idea why, it seems fine, when trying again.
> 
> This change fixed it probably; [e1ecab3](https://github.com/Foxboron/ssh-tpm-agent/commit/e1ecab37dacce5631b73ab2309e9e13e65acd964)

aha, yep the problem was that using `go install ...@latest` (and AUR) installs 0.1.0, since release candidates are not actual releases. I created the unit before I upgraded to 1.0.0-rc2.

**Comment by Foxboron** [comment]
> aha, yep the problem was that I installed 0.1.0 with go install ...@latest (or AUR), since release candidates are not actual releases. I created the unit before I upgraded to 1.0.0-rc2.

Ohh, I was not aware that `go` would be that clever and ignore the tag.

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ install: $(BINS)
 	@install -dm755 $(DESTDIR)$(LIBDIR)/systemd/system
 	@install -dm755 $(DESTDIR)$(LIBDIR)/systemd/user
 	@DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) bin/ssh-tpm-hostkeys --install-system-units 
-	@TEMPLATE_BINARY=1 DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) bin/ssh-tpm-agent --install-user-units --install-system
+	@TEMPLATE_BINARY=/usr/bin/ssh-tpm-agent DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) bin/ssh-tpm-agent --install-user-units --install-system
```
Comment (Foxboron) [inline Makefile:27]:
```suggestion
	@TEMPLATE_BINARY=$(BINDIR)/ssh-tpm-agent DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) bin/ssh-tpm-agent --install-user-units --install-system
```

**Review by Foxboron — APPROVED** [review]

**Comment by Foxboron** [comment]
Generally for the future, it would be better to have the descriptions explaining "why" in the commits themselves, then we won't be at the mercy of github to contexualize the commits :)

Other then that it's fine. I can amend the `BINDIR` change.

**Comment by Foxboron** [comment]
Merged with https://github.com/Foxboron/ssh-tpm-agent/commit/7f5985f2394c5f86eef6fe91a850833e3e6110c1

**Pull request closed without merging**